src/java/main/org/apache/zookeeper/server/quorum/Learner.java

```diff
@@ -364,10 +367,12 @@ else if (qp.getType() == Leader.SNAP) {
             
             long lastQueued = 0;
 
-            // in V1.0 we take a snapshot when we get the NEWLEADER message, but in pre V1.0
+            // in Zab V1.0 (ZK 3.4+) we might take a snapshot when we get the NEWLEADER message, but in pre V1.0
             // we take the snapshot at the UPDATE, since V1.0 also gets the UPDATE (after the NEWLEADER)
             // we need to make sure that we don't take the snapshot twice.
-            boolean snapshotTaken = false;
+            boolean isPreZAB1_0 = true;
+            //If we are not going to take the snapshot be sure the edits are not applied in memory
+            boolean writeToEditLog = !snapshotNeeded;
```

Comment:
Sorry about that I am still a bit new at the internal terminology.  I will update it.